designs/switch-prometheus-to-opentelemetry.md

```diff
@@ -0,0 +1,172 @@
+Switch Prometheus to OpenTelemetry
+===================
+
+## Motivation
+
+Today, controller-runtime already provides metrics accessible through the
+Prometheus client, and exposed on the `/metrics` endpoint. However, this forces
+components to use Prometheus, when they may want to monitor their app another
+way, not necessarily with a pull system.
+
+## Goals
+
+* Provide a generic way to send metrics, so they can be exported virtually anywwhere.
+* Do not introduce existing metrics changes for users
+
+## Proposal
+
+We will entirely remove the current metrics server. If users need to run
+Prometheus, they will still be able to do so after configuring OpenTelemetry by
+using the included exporter.
+
+We will configure a global meter so we can send metrics from anywhere within
+the component. That meter will be set when needed, and not at boot time, so
+users can set their own `MeterProvider`  and specify where they need to exporte
```

Comment:
What exactly is a `Meter`? The equivalent of a prometheus Registry?

Reply:
A `MeterProvider` holds the exporter configuration (whether the metrics are exported to Prometheus for example), and each `Meter` represents a different intrumentation library (since all controller-runtime is the same instrumentation library, we use the same across the entire repository).

See https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/metrics/api.md#meter-provider